controllers/operatorpolicy_controller.go

```diff
@@ -133,9 +146,96 @@ func (r *OperatorPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 
 	// FUTURE: more resource checks
 
+	// handle catalogsource
+	if err := r.handleCatalogSource(ctx, policy); err != nil {
+		OpLog.Error(err, "Error handling CatalogSource")
+
+		return reconcile.Result{}, err
+	}
+
 	return reconcile.Result{}, nil
 }
 
+func (r *OperatorPolicyReconciler) handleCatalogSource(
+	ctx context.Context,
+	policy *policyv1beta1.OperatorPolicy,
+) error {
+	watcher := opPolIdentifier(policy.Namespace, policy.Name)
+
+	desiredSub, err := buildSubscription(policy)
+	if err != nil {
+		return fmt.Errorf("error building Subscription: %w", err)
+	}
+
+	// Retrieve Subscription
+	foundSub, err := r.DynamicWatcher.Get(watcher, subscriptionGVK, desiredSub.Namespace, desiredSub.Name)
+	if err != nil {
+		return fmt.Errorf("error getting Subscription: %w", err)
+	}
```

Comment:
I think it will be better if we use a subscription returned by `handleSubscription`. I'll share a snippet that we can either bring into this PR, or it might be ok to accept this as-is and adjust it in another PR.

Reply:
I think @JeffeyL's PR was going to modify `handleSubscription` to return the subscription. I wanted to draft this one up to get some reviews first but I can always adjust it later.